File: PHP/Legacy/Path.php
#! /usr/bin/env php
<?php

require_once("Utility.php");

function Path() {

	println(<<<"Path"
		ホームディレクトリ:       {$i(posix_getpwuid(posix_geteuid())["dir"])}
		カレントディレクトリ:     {$i(getcwd())}

		このファイルのフルパス:   {$i(__FILE__)}
		このファイルの実行パス:   {$argv[0]}
		このファイルの名前:       {$i(basename(__FILE__))}
		このファイルのある場所:   {$i(__DIR__)}
	Path);

	pnl(2);

}

if (running_directly(__FILE__)) Path();

?>

File: Ruby/Legacy/Path.rb
#! /usr/bin/env ruby

require_relative "Utility.rb"

def Path

	println """
		ホームディレクトリ:       #{Dir.home}
		カレントディレクトリ:     #{Dir.pwd}

		このファイルのフルパス:   #{File.realpath(__FILE__)}
		このファイルの実行パス:   #{__FILE__} = $0
		このファイルの名前:       #{File.basename(__FILE__)}
		このファイルのある場所:   #{__dir__}
	"""

	pnl 2

end

Path() if runningDirectly(__FILE__)